tests/file_writer/test_recording.py

```diff
@@ -1724,3 +1660,96 @@ def test_FileWriterProcess__deletes_recorded_stim_data_after_stop_time(
         )
 
         this_file.close()
+
+
+def test_FileWriterProcess__upload_thread_gets_added_to_container_after_all_files_get_finalized(
+    four_board_file_writer_process, mocker
+):
+    update_customer_settings_command = copy.deepcopy(GENERIC_UPDATE_CUSTOMER_SETTINGS)
+    update_customer_settings_command["config_settings"]["auto_delete_local_files"] = False
+    update_customer_settings_command["config_settings"]["auto_upload_on_completion"] = True
+    file_writer_process_ready_for_upload = file_writer_process_with_closed_h5_files_for_upload(
+        four_board_file_writer_process=four_board_file_writer_process,
+        update_customer_settings_command=update_customer_settings_command,
+    )
+    file_writer_process = file_writer_process_ready_for_upload["fw_process"]
+    to_main_queue = file_writer_process_ready_for_upload["to_main_queue"]
+    mocker.patch.object(file_uploader.ErrorCatchingThread, "errors", autospe=True, return_value=True)
+    mocker.patch.object(
+        file_uploader.ErrorCatchingThread, "get_error", autospe=True, return_value="mocked_error"
+    )
+
+    invoke_process_run_and_check_errors(file_writer_process, num_iterations=6)
+    # pylint: disable=protected-access
+    assert (
+        file_writer_process._customer_settings == GENERIC_UPDATE_CUSTOMER_SETTINGS["config_settings"]
+    )  # pylint: disable=protected-access
+    assert (
+        file_writer_process._is_finalizing_files_after_recording() is False
+    )  # pylint: disable=protected-access
+    assert len(file_writer_process._open_files[0].keys()) == 0  # pylint: disable=protected-access
+    assert len(file_writer_process._upload_threads_container) == 1  # pylint: disable=protected-access
+    assert (
+        file_writer_process._upload_threads_container[0]["failed_upload"] is False
+    )  # pylint: disable=protected-access
+    assert (
+        file_writer_process._upload_threads_container[0]["customer_account_id"] == "test_customer_id"
+    )  # pylint: disable=protected-access
+    assert (
+        file_writer_process._upload_threads_container[0]["auto_delete"] is False
+    )  # pylint: disable=protected-access
+    assert (
+        file_writer_process._upload_threads_container[0]["file_name"] == "MA200440001__2020_02_09_190935"
+    )  # pylint: disable=protected-access
+
+    invoke_process_run_and_check_errors(file_writer_process, num_iterations=8)
+
+    assert to_main_queue[-1]["communication_type"] == "update_upload_status"
```

Comment:
Indexing into the queue like this only works because a `TestingQueue` is used here which is just a modified deque, so this test would fail if someone changed it to a real queue. There's a helper function called `drain_queue` which will give you a list of all the items in the queue which is useful when you just want to assert something on the last item to be put into a queue